providers/apis/defi/uniswapv3/fetcher.go

```diff
@@ -192,17 +193,23 @@ func (u *PriceFetcher) Fetch(
 		pools[i] = pool
 	}
 
-	// Batch call to the EVM.
-	if err := u.client.BatchCallContext(ctx, batchElems); err != nil {
-		u.logger.Debug(
-			"failed to batch call to ethereum network for all tickers",
-			zap.Error(err),
-		)
+	// process 10 tickers at a time
+	const batchSize = 10
+	batchChunks := slices.Chunk(batchElems, batchSize)
 
-		return types.NewPriceResponseWithErr(
-			tickers,
-			providertypes.NewErrorWithCode(err, providertypes.ErrorAPIGeneral),
-		)
+	for _, chunk := range batchChunks {
```

Comment:
We should probably just parallelize this. It isn't too much more complicated, and this is one of the most common ways for latency to get out of control.

```go
	eg, egCtx := errgroup.WithContext(ctx)
	for _, chunk := range batchChunks {
		chunk := chunk
		eg.Go(func() error {
			// Batch call to the EVM.
			if err := u.client.BatchCallContext(egCtx, chunk); err != nil {
				u.logger.Debug(
					"failed to batch call to ethereum network for all tickers",
					zap.Error(err),
				)

				return err
			}

			return nil
		})
	}

	if err := eg.Wait(); err != nil {
		return types.NewPriceResponseWithErr(
			tickers,
			providertypes.NewErrorWithCode(err, providertypes.ErrorAPIGeneral),
		)
	}
```

Reply:
If you do this, you should probably set a limit of something like 5 for the error group.